### DISCUSSION TITLE
Is it possible to have columns with same field

### DISCUSSION THREAD

**Discussion by KaushikEDS** [IgniteUI/igniteui-angular#12111] [General]
We have created a feature for our clients where they can create their own Column groups.

We are facing problems when we add a column to a group and then have the same column in the grid as an individual column.
For example **Group Column A**  has children **Child A**, **Child B**, **Child C**. Also, we have a normal column **Child B** in the grid.
![image](https://user-images.githubusercontent.com/79090369/186475353-12b0900c-e10d-445c-815a-3ba48e2f5c7d.png)

I think the issue is that both columns have the same field property.

So far we have found issues with grid summaries and grid selectedCells where we get data for only the leftmost column. But there also might be other issues that we haven't found yet.

Is there a way to keep both columns with the same field?

**Comment by MonikaKirkova** [comment]
Hello @KaushikEDS ,

I have been looking into your question and I do not recommend using the same field for two columns because in some scenarios there could be side effects in case the functionality depends on the column field.
However, I have created a sample having one column in a column group and one individual and on my side I cannot observe any issues with the selection and summaries. 
Please test [the sample](https://stackblitz.com/edit/angular-9dflbx?file=src/app/grid/multi-column-headers/multi-column-headers.html) on your side and let me know how it behaves. If this is not an accurate demonstration of what you are trying to achieve please feel free to modify it and send it back to me along with steps to reproduce.

**Comment by KaushikEDS** [comment]
@MonikaKirkova 
I tried opening the sample on stackblitz, but it seems to have some issues. 
![image](https://user-images.githubusercontent.com/79090369/186653569-23f785bd-34ec-4316-b390-008395f9d2a3.png)
****


I downloaded it on my local
In your sample 
When we select cells in the right column the grid.selectedCells property shows empty array.
![image](https://user-images.githubusercontent.com/79090369/186660181-cc99fb9d-8d6d-4be9-b410-78b9a77b8485.png)

When we select the left column the grid.selectedCells show cells of the right column and itself.
![image](https://user-images.githubusercontent.com/79090369/186663865-5558b9d4-c286-475a-a4ea-ce12860118e5.png)

Edited sample :-

[angular-sample.zip](https://github.com/IgniteUI/igniteui-angular/files/9425062/angular-sample.zip)

**Comment by MonikaKirkova** [comment]
Hello @KaushikEDS,

I have investigated this further and I have determined that the selected cells are accessed based on the cells in the row. However, this depends on the column field and when the field is not unique, as it is recommended, the cells in both columns are stored as selected.
Depending on your requirement, in order for all features to work as expected, what I could suggest is setting an unbound column and displaying the same value as CompanyName by using a formatter function.
The second option is to modify the data and to create a different field with the same values, which could be handled on application level.
